bpf/utils.h

```diff
@@ -211,7 +218,7 @@ static inline bool check_and_do_flow_filtering(flow_id *id, u16 flags, u32 drop_
             // we have no matching rules so we update global counter for flows that are not matched by any rule
             increase_counter(FILTER_NOMATCH);
             // we have accept rule but no match so we can't let mismatched flows in the hashmap table.
-            if (action == ACCEPT || action == MAX_FILTER_ACTIONS) {
+            if (action == ACCEPT) {
```

Comment:
I guess I see why you removed this, I think it fixes an issue but introduces a new one. In case of no CIDR match, action is neither ACCEPT nor REJECT and previously we would view that as a DROP; now you change that to be a KEEP. I think neither of those options are valid, we need to inspect the filters to decide whether it should be DROP or KEEP.
If users have defined any ACCEPT rules, then a no-match should be DROP
If users have defined any REJECT rules, then a no-match should be KEEP

... and if users have defined a mix of ACCEPT and REJECT, I think it just doesn't make sense because we don't know what to say when no rule is matched?

So at the end of the day, I think it's an API issue: we shouldn't allow to set ACCEPT/REJECT action _per rule_, it should be a single setting for all rules.

Reply:
we should be the one define what is the default behavior when no CIDR match happens which is when we get the special action value, in that case I feel keeping the default as before to disallow is more accurate , nothing wrong with the API IMO, we just need to define what is the nomatch action behavior

Reply:
I don't agree, there's no point having ALLOW by default if you define ALLOW rules, and there's no point having DENY by default if you define DENY rule. It's an API problem.
If we keep as it is today, then there's a whole range of things you can't express, e.g:

```
Filter: { 1.2.3.4/32: REJECT }
```

You would expect a flow like that to be kept: src=6.6.6.6, dst=8.8.8.8
But here it would be dropped because action == MAX_FILTER_ACTIONS

Reply:
yeah u need to add Allow 0.0.0.0/0

Reply:
When you add Allow 0.0.0.0/0, you fall into another problem: since the src_ip is checked before the dst_ip, this global-allow will take precedence on a flow like `src=6.6.6.6, dst=1.2.3.4` even though there is the reject rule for that IP that in theory should take precedence.
Maybe then you need to always run both lookups for source and dest, and not check dest only when src has no match.

I think we should be more clear on what to expect with this feature, I've started a doc for acceptance criteria: https://docs.google.com/document/d/1YUVcpCdYxRACRVrqEOwir6upMyla5RvvUsZmgP4WvIs/edit?tab=t.0 - I think there's already a couple of issues - things that are not covered